src/Lithnet.GoogleApps.MA/AttributeAdapters/PatchableAdapterPropertyValue.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.MetadirectoryServices;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lithnet.GoogleApps.MA.AttributeAdapters
+{
+    internal class PatchableAdapterPropertyValue : AdapterPropertyValue
+    {
+
+        public override bool CanPatch(KeyedCollection<string, AttributeChange> changes)
+        {
+            // Patching appeared to not be working correctly using AdapterPropertyValue for Course, so override
```

Comment:
Have you stumbled on an underlying bug we should fix, or is this a condition specific to courses API that doesn't work with existing logic?

Reply:
Yes, during development I had a feeling that Full updates are done more often than neccessary. 

I may be reading the code wrong, but it seems as the current CanPatch method as currently used requires all properties of the schema to be Patchable. If not, it does a full update.

The code is called from here:

Reply:
public bool CanPatch(KeyedCollection<string, AttributeChange> changes)
        {
            return this.SupportsPatch && this.AttributeAdapters.All(t => t.CanPatch(changes));
        }

In AdapterPropertyValue the code looks like this:
 public virtual bool CanPatch(KeyedCollection<string, AttributeChange> changes)
        {
            return this.SupportsPatch;
        }

From what I can deduce this requires all attributes in the schema to be patchable. Otherwise it'll return false.

Reply:
I'm think your code is correct. I'm going to delete the patchable adapter, and integrate your changes into the original class